### ISSUE TITLE
[Contributors Wanted] Add `subset="both"` functionality to `{image|text}_dataset_from_directory()`

### ISSUE THREAD

**Issue by fchollet** [keras-team/keras#16384]
Current if you want to load a dataset using `{image|text}_dataset_from_directory()` and split it into a training and validation set, you do:

```
train_ds = X_dataset_from_directory(..., validation_split=0.5, seed=1337, subset="training")
val_ds = X_dataset_from_directory(..., validation_split=0.5, seed=1337, subset="validation")
```

We'd like to introduce the following flow:

```
train_ds, val_ds = X_dataset_from_directory(..., validation_split=0.5, subset="both")
```

This applies to both `image_dataset_from_directory()` and `text_dataset_from_directory()`.

If you're interested in contributing, please open a draft PR!

**Comment by thisisbhavin** [comment]
Would the previous methods stay?

**Comment by fchollet** [comment]
> Would the previous methods stay?

Yes. This is a pure addition. In general any change must be backwards compatible.

**Comment by fchollet** [comment]
This has now been implemented! 👍
